### DISCUSSION TITLE
Rework future documentation to add source of diagrams in repo as plain text

### DISCUSSION THREAD

**Discussion by carmel4a** [MaSzyna-Reloaded/MaSzyna-API-wrapper#56] [Ideas]
.png is final format, that we cannot modify, it would be better to add plain `plantUML/draw.io/other` files to repo, then export them to `.png` during CI process.

**Comment by marcinn** [comment]
It would be nice to automatically render pngs at building the site. Alternatively render diagrams at runtime, i.e. by using Mermaid code blocks

* https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-diagrams
* http://mermaid.js.org/intro/

There is plugin for Jekyll https://github.com/mutcher/jekyll-drawio, but it uses drawio online viewer and ... it is not working for me :(
![image](https://github.com/user-attachments/assets/eb04dadc-14e4-4e8f-a69a-125f29b7b7fa)


Or after switching from Jekyll  to Hugo:
* GoAT, Mermaid,  https://gohugo.io/content-management/diagrams/

I'm afraid that drawio will be problematic to render. We can add drawio files to the repo, of course, but still there will be manual work required to export them as svg or png. Mermaid seems to be common solution. GoAT seems to be a kind of WYSIWYG markup, but probably requires more ascii art skills https://github.com/blampe/goat
